### PR TITLE
Stylus -> SCSS

### PR DESCRIPTION
**Pull request by christianmemije**
### Summary

Converts stylus to scss.

TODO: Prettify scss. Add stylint.

…

### Reviewer guidance

Review

…

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,13 +91,13 @@
...
     "mkdirp": "^0.5.1",
-    "node-sass": "^4.5.0",
+    "node-sass": "^4.9.1",
     "optimize-css-assets-webpack-plugin": "^4.0.0",
     "postcss-loader": "^2.0.6",
     "prettier": "^1.10.2",
     "responselike": "^1.0.2",
     "rewire": "^2.5.2",
-    "sass-loader": "^6.0.6",
+    "sass-loader": "^7.0.3",
```
Comment (jonboiser) [inline package.json]: If stylus isn't used anywhere, could you remove the stylus tooling from here and the webpack configurations?

File: frontend_build/src/prettier-frontend.js [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+const prettier = require('prettier');
+const compiler = require('vue-template-compiler');
 
 const errorOrChange = 1;
 const noChange = 0;
+/*
+  Modifications to match our component linting conventions.
+  Surround style and script blocks by 2 new lines and ident.
+*/
+function identAndAddNewLines(str) {
```
Comment (indirectlylit) [inline frontend_build/src/prettier-frontend.js]: `indent` - ident sounds like "identity"

File: frontend_build/src/prettier-frontend.js [hunk 3]
```diff
@@ -24,29 +33,53 @@ function prettierFrontend({ file, write, encoding = 'utf-8', prettierOptions })
...
+            formattedJs = identAndAddNewLines(formattedJs);
+            formatted = source.replace(source.slice(scriptStart, scriptEnd), formattedJs);
+          }
+
+          // Format style blocks
+          for (let i = 0; i < vueComponent.styles.length; i++) {
+            // Reparse to get updated line numbers
+            const styleBlock = compiler.parseComponent(formatted).styles[i];
+            // Skip if not scss or css
+            if (styleBlock.lang !== 'scss') {
```
Comment (indirectlylit) [inline frontend_build/src/prettier-frontend.js]: the comment says "Skip if not scss or css" but it only checks `scss`

File: frontend_build/src/webpack.config.base.js [hunk 4]
```diff
@@ -172,13 +172,13 @@ module.exports = {
   },
   plugins: [
     new PrettierFrontendPlugin({
-      extensions: ['.js', '.vue'],
+      extensions: ['.js', '.vue', '.scss'],
```
Comment (indirectlylit) [inline frontend_build/src/webpack.config.base.js:174]: 🎉

File: kolibri/core/assets/src/views/app-bar.vue [hunk 5]
```diff
@@ -209,14 +209,14 @@
...
-    background-size: cover;
-  }
+  // .app-bar-title-icon {
+  //   background: $app-bar-title-icon;
+  //   height: $app-bar-title-icon-height;
+  //   width: $app-bar-title-icon-height;
+  //   display: inline-block;
+  //   vertical-align: middle;
+  //   background-size: cover;
+  // }
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/app-bar.vue]: was this just used in VF 0.6?

we can probably delete it, assuming we'll need to re-implement in the future using another mechanism

File: kolibri/core/assets/src/views/immersive-full-screen.vue [hunk 6]
```diff
@@ -65,8 +65,7 @@
   .top-bar {
     height: 60px;
     background-color: $core-action-dark;
-    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2),
-                0 6px 20px 0 rgba(0, 0, 0, 0.19);
+    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/immersive-full-screen.vue]: ah bummer - I liked the stacked formatting. guess prettier doesn't

**Review by indirectlylit — COMMENTED** [review]
looking good

File: frontend_build/src/prettier-frontend.js (hunk 2 above)
Comment (christianmemije) [reply frontend_build/src/prettier-frontend.js]: Oops typo.

File: kolibri/core/assets/src/views/app-bar.vue (hunk 5 above)
Comment (christianmemije) [reply kolibri/core/assets/src/views/app-bar.vue]: Yes it was a hack to get the logo to appear in the app-bar. But in SCSS, undefined variables are not allowed.

**Comment by indirectlylit** [comment]
sorry looks like yarn.lock has a conflict

File: kolibri/plugins/user/assets/src/views/sign-in-page/index.vue [hunk 7]
```diff
@@ -395,107 +395,126 @@
...
+    text-align: center;
+    width: 100%;
+    height: 100%;
+    display: table;
+  }
+
+  .main-row {
+    display: table-row;
+    background-color: $core-action-normal;
+    background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7)), url(./background.jpg);
```
Comment (indirectlylit) [inline kolibri/plugins/user/assets/src/views/sign-in-page/index.vue]: this line is giving me

```
>     /Users/d/Projects/learning_equality/repos/k011x/kolibri/plugins/user/assets/src/views/sign-in-page/index.vue
      416:1  error  Line 416 exceeds the maximum line length of 100  max-len
    
    ✖ 1 problem (1 error, 0 warnings)
```

Any chance this is happening due to `indentAndAddNewLines`?

Comment (christianmemije) [reply kolibri/plugins/user/assets/src/views/sign-in-page/index.vue]: I think this is a prettier scss bug. This line below for example, has 102 chars but the print-width is set to 100.

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATAlgN2AHSgAnwCMBDMAawHMAnCAVylQFp0BbEyuJfAG3SjglqTGiQwIYACmqVSkgAwAafEpXL5AOgDsASmUy5qo+u069+OtR6SNAelIUa9cuUYaAVgAdKOgNx4AXxBFEAhPGHRoAGdkUCFaAHcABSEEGJQSTAh0VGCQImoycjgYAGVPMn5KZBhqOjgQ-ii4ahgkwsp2ZAAzEh5mkPcogA8AIUKKEtKSVjgAGX44Hr6BkArqZupkECiwKJiQz2p+GAB1HJgAC2QARnklNdpm08LPbaO4TcwlkOo4AEc6Og-u0OF0kL1+g0dnBWOganVoVEqjw4ABFOgQeDLKEhGAkIjnVBXZAAJjxhXQfCglAAwhBWOCQFBoD8QHRmgAVAnpSHNAIBIA

**Review by indirectlylit — APPROVED** [review]
💥

**Pull request merged**